src/convenience.jl

```diff
@@ -94,38 +94,46 @@ function cluster!(
     connection;
     input_database_schema,
     table_names = Dict("profiles" => input_profile_table_name),
+    layout,
   )
 
   if input_database_schema != ""
     input_profile_table_name = "$input_database_schema.$input_profile_table_name"
   end
-  df = DuckDB.query(
+
+  profiles = DuckDB.query(
     connection,
     "SELECT * FROM $input_profile_table_name
     ",
   ) |> DataFrame
-  # If user provided custom layout expecting different column names, rename now so downstream honors them
-  if layout.timestep != :timestep && hasproperty(df, :timestep)
-    rename!(df, :timestep => layout.timestep)
-  end
-  if layout.value != :value && hasproperty(df, :value)
-    rename!(df, :value => layout.value)
+
+  add_optional_columns!(profiles; layout)
```

Comment:
Pair review comment. We can avoid including `scenario` because we can populate with defaults later